crates/sui-orchestrator/src/measurement.rs

```diff
@@ -0,0 +1,396 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    collections::HashMap,
+    fs,
+    io::BufRead,
+    path::{Path, PathBuf},
+    time::Duration,
+};
+
+use prettytable::{format, row, Table};
+use prometheus_parse::Scrape;
+use serde::{Deserialize, Serialize};
+
+use crate::{benchmark::BenchmarkParameters, settings::Settings};
+
+/// The identifier of prometheus latency buckets.
+type BucketId = String;
+
+/// A snapshot measurement at a given time.
+#[derive(Serialize, Deserialize, Default, Clone)]
+pub struct Measurement {
+    /// Duration since the beginning of the benchmark.
+    timestamp: Duration,
+    /// Latency buckets.
+    buckets: HashMap<BucketId, usize>,
```

Comment:
what is the granularity of the buckets?

Reply:
We don't choose it. We parse whatever buckets are set by the load generator (in a generic way, so if the load generator changes the buckets granularity nothing should break here). The current buckets of the land generator are set here: https://github.com/MystenLabs/sui/blob/4e55796f6c78ed322b337ea3b07e6dc1e47ea2aa/crates/sui-benchmark/src/drivers/bench_driver.rs#L54